src/main/java/codechicken/nei/recipe/StackInfo.java

```diff
@@ -223,7 +224,9 @@ public static ItemStack getItemStackWithMinimumDamage(ItemStack[] stacks) {
 
         if (stacks.length > 1) {
             for (ItemStack stack : stacks) {
-                if (stack.getItem() != null && !ItemInfo.isHidden(stack) && stack.getItemDamage() < damage) {
+                if (stack.getItem() != null && !ItemInfo.isHidden(stack)
+                        && (stack.getItemDamage() < damage || stack.getItemDamage() == damage
+                                && Item.getIdFromItem(stack.getItem()) < Item.getIdFromItem(result.getItem()))) {
```

Comment:
This is kind of hard to parse. This change would work to clarify the logic, but could it maybe be split up a bit more instead?
```suggestion
                        && (stack.getItemDamage() < damage || (stack.getItemDamage() == damage
                                && Item.getIdFromItem(stack.getItem()) < Item.getIdFromItem(result.getItem())))) {
```